build-a-react-native-application-for-ios-and-android-from-start-to-finish/notes/06-accept-user-input-with-react-native-textinput.md

```diff
@@ -0,0 +1,56 @@
+# Accept User Input with React Native TextInput
+
+**[📹 Video](https://egghead.io/lessons/react-native-accept-user-input-with-react-native-textinput)**
+
+## Text input
+
+The `<TextInput />` component can be used anytime we want the user to input data. It is similar to a HTML `<input />`, but does not come with any default styling. May need to add a border or background color to distinguish where it is.
+
+The `onChangeText` prop can be used to run a function when the user changes the text in a `<TextInput />`. This is similar to the `onChange` prop in React, however, the argument passed to `onTextChange` is the value itself. Much more convenient than digging down from the event.
+
+```
+<TextInput
+  value={this.state.username}
+  onChange={value => this.setState({ username: value })}
+/>
+```
+
+## No shorthand properties
+In CSS there are a few convenient shorthand properties for styling a collection of related things - such as `border: 1px dotted green;` or `padding: 1rem 2rem;`. Unfortunately these shorthands do not exist in React Native as all properties must be single values. Border properties must be specified as separate rules.
```

Comment:
```suggestion
In CSS there are a few convenient shorthand properties for styling a collection of related things - such as `border: 1px dotted green;` or `padding: 1rem 2rem;`. **Unfortunately, these shorthands do not exist in React Native as all properties must be single values.** Border properties must be specified as separate rules.
```

Reply:
Added bolding to this statement as that seems important 😄  we do a lot of progressive summarization at egghead! https://fortelabs.co/blog/progressive-summarization-a-practical-technique-for-designing-discoverable-notes/

(these notes are a form of prog. sum.)